src/envoy/http/backend_routing/filter.cc

```diff
@@ -39,20 +47,25 @@ FilterHeadersStatus Filter::decodeHeaders(
   // NOTE: this shouldn't happen in practice because Path Matcher filter would
   // have already rejected the request.
   if (operation.empty()) {
-    ENVOY_LOG(debug, "No operation found from DynamicMetadata");
-    return FilterHeadersStatus::Continue;
+    rejectRequest(Envoy::Http::Code::InternalServerError,
+                  "No operation found from DynamicMetadata",
+                  RcDetails::get().BadRequest);
+    return FilterHeadersStatus::StopIteration;
```

Comment:
I think the design of this filter is:  passive.   If I don't not handle it, pass through.  In this way, it is easier to config this filter, it doesn't need to know all request patterns.  It only needs to know the request patterns that it need to act.

It makes this filter safer to be added.

Reply:
I see, then let's pass through for this case and error for the other cases (that shouldn't happen).

Also seems like the integration tests didn't catch my error. I will attempt to make a follow-up PR that tests this, if we want to support both sidecar and dynamic routing modes at the same time.